core/src/test/java/org/apache/iceberg/io/InMemoryOutputFile.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.io;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.Objects;
+import java.util.UUID;
+
+public class InMemoryOutputFile implements OutputFile {
+
+  private final String location;
+
+  private boolean exists;
+  private ByteArrayOutputStream contents;
+
+  public InMemoryOutputFile() {
+    this("memory:" + UUID.randomUUID());
+  }
+
+  public InMemoryOutputFile(String location) {
+    this.location = Objects.requireNonNull(location, "location is null");
+  }
+
+  @Override
+  public PositionOutputStream create() {
+    if (exists) {
+      throw new RuntimeException("Already exists");
```

Comment:
Can we throw `AlreadyExistsException` to match the current `LocalOutputFile#create()` method?

https://github.com/apache/iceberg/blob/e45c19e1d02a130d33ac886f676dfb141708090d/api/src/main/java/org/apache/iceberg/Files.java#L57-L59

Reply:
Thanks, indeed this is what interface requires me to do